File: ruby/spec/slug.spec.rb
require 'enotype'

describe 'slug' do
  context 'with \'eno-lang-article\'' do
    it 'produces the expected result' do
      expect(Enotype::slug('eno-lang-article')).to eq('eno-lang-article')
    end
  end
  
  context 'with \'eno_lang_article\'' do
    it 'produces the expected result' do
      expect(Enotype::slug('eno_lang_article')).to eq('eno_lang_article')
    end
  end
  
  context 'with \'eno-lang-article!\'' do
    it 'raises an error' do
      expect { Enotype::slug('eno-lang-article!') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'%eno-lang-article\'' do
    it 'raises an error' do
      expect { Enotype::slug('%eno-lang-article') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'eno lang article\'' do
    it 'raises an error' do
      expect { Enotype::slug('eno lang article') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'enö-läng-ärticle\'' do
    it 'raises an error' do
      expect { Enotype::slug('enö-läng-ärticle') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'énó-láng-ártíclé\'' do
    it 'raises an error' do
      expect { Enotype::slug('énó-láng-ártíclé') }.to raise_error(RuntimeError)
    end
  end
end

File: php/spec/slug.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('slug', function() {
  describe('with \'eno-lang-article\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::slug('eno-lang-article'))->toEqual('eno-lang-article');
    });
  });
  
  describe('with \'eno_lang_article\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::slug('eno_lang_article'))->toEqual('eno_lang_article');
    });
  });
  
  describe('with \'eno-lang-article!\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::slug('eno-lang-article!'); })->toThrow();
    });
  });
  
  describe('with \'%eno-lang-article\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::slug('%eno-lang-article'); })->toThrow();
    });
  });
  
  describe('with \'eno lang article\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::slug('eno lang article'); })->toThrow();
    });
  });
  
  describe('with \'enö-läng-ärticle\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::slug('enö-läng-ärticle'); })->toThrow();
    });
  });
  
  describe('with \'énó-láng-ártíclé\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::slug('énó-láng-ártíclé'); })->toThrow();
    });
  });
});